crates/tabby-common/src/config.rs

```diff
@@ -50,13 +50,17 @@ impl RepositoryConfig {
             let path = self.git_url.strip_prefix("file://").unwrap();
             path.into()
         } else {
-            repositories_dir().join(to_filename(&self.git_url))
+            repositories_dir().join(Self::name(&self.git_url))
         }
     }
 
     pub fn is_local_dir(&self) -> bool {
         self.git_url.starts_with("file://")
     }
+
+    pub fn name<S: AsRef<str>>(s: S) -> String {
+        filenamify(s)
+    }
 }
```

Comment:
can this be
```
pub fn name(&self) -> String {
  filenamify(self.git_url)
}

```

Reply:
this method is also used in `load_dataset` to generate `DatasetKey`, current implementation is:
```
let key = DatasetKey {
    local_name: RepositoryConfig::name(file.git_url.as_str()),
    rel_path: file.filepath.clone(),
};
```

If we define `name` like what you described, above implementation must be changed to:
```
let conf = RepositoryConfig { git_url: file.git_url.clone() };
let key = DatasetKey {
    local_name: conf.name(),
    rel_path: file.filepath.clone(),
};
```

I'm ok with it, what's your opinion?

Reply:
I think there's a missing mapping between git_url to RepositoryConfig - I suggest in load_dataset we simply fetch `Config::load()` and build a map between git url to repository config, then fetch the name.

In future we might introduce a configurable `name` field inside of `RepositoryConfig`, this refactor will also make it easier to happen.